py-polars/polars/expr/datetime.py

```diff
@@ -1423,6 +1423,83 @@ def replace_time_zone(
             self._pyexpr.dt_replace_time_zone(time_zone, ambiguous._pyexpr)
         )
 
+    def to_naive_local(
+        self,
+        tz: IntoExpr,
+    ) -> Expr:
+        """
+        Convert to local datetime in given time zone.
+
+        Parameters
+        ----------
+        tz
+            Time zone for the `Datetime` expression.
+
+        Returns
+        -------
+        Expr
+            Expression of data type :class:`DateTime`.
+
+        Examples
+        --------
+        >>> from datetime import datetime
+        >>> df = pl.DataFrame(
+        ...     {
+        ...         "date_col": [datetime(2020, 10, 10)] * 3,
+        ...         "timezone": ["Europe/London", "Africa/Kigali", "America/New_York"],
+        ...     }
+        ... ).with_columns(pl.col("date_col").dt.replace_time_zone("UTC"))
+        >>> df.with_columns(
+        ...     pl.col("date_col").dt.to_naive_local("timezone").alias("local_dt")
+        ... )
+        shape: (3, 3)
+        ┌─────────────────────────┬──────────────────┬─────────────────────┐
+        │ date_col                ┆ timezone         ┆ local_dt            │
+        │ ---                     ┆ ---              ┆ ---                 │
+        │ datetime[μs, UTC]       ┆ str              ┆ datetime[μs]        │
+        ╞═════════════════════════╪══════════════════╪═════════════════════╡
+        │ 2020-10-10 00:00:00 UTC ┆ Europe/London    ┆ 2020-10-10 01:00:00 │
+        │ 2020-10-10 00:00:00 UTC ┆ Africa/Kigali    ┆ 2020-10-10 02:00:00 │
+        │ 2020-10-10 00:00:00 UTC ┆ America/New_York ┆ 2020-10-09 20:00:00 │
+        └─────────────────────────┴──────────────────┴─────────────────────┘
+
+        """
+        local_tz = parse_as_expression(tz, str_as_lit=False)
```

Comment:
I don't think there's a general rule about whether a string should be parsed as literal or expression, but in other functions with time zones, the time zone is parsed as a literal - so for consistency, could we do that here too please?

`.dt.to_naive_local('Asia/Kathmandu')` is more ergonomic than `.dt.convert_time_zone('Asia/Kathmandu').dt.datetime()` so people might reasonably want to use this with just a single time zone

Reply:
But then if you want to pass a column, you have to do pl.col("tz"), right?